src/main/java/org/opensearch/ad/ml/ModelManager.java

```diff
@@ -838,33 +839,41 @@ public ThresholdingResult getAnomalyResultForEntity(
     }
 
     public ThresholdingResult score(double[] feature, String modelId, ModelState<EntityModel> modelState) {
+        ThresholdingResult result = null;
         EntityModel model = modelState.getModel();
         if (model == null) {
-            return new ThresholdingResult(0, 0, 0);
-        }
-        RandomCutForest rcf = model.getRcf();
-        ThresholdingModel threshold = model.getThreshold();
-        if (rcf == null || threshold == null) {
-            return new ThresholdingResult(0, 0, 0);
-        }
-
-        // clear feature not scored yet
-        Queue<double[]> samples = model.getSamples();
-        while (samples != null && samples.peek() != null) {
-            double[] recordedFeature = samples.poll();
-            double rcfScore = rcf.getAnomalyScore(recordedFeature);
-            rcf.update(recordedFeature);
-            threshold.update(rcfScore);
-        }
+            result = new ThresholdingResult(0, 0, 0);
+        } else if (model.getErcf().isPresent()) {
+            ExtendedRandomCutForest ercf = model.getErcf().get();
+            Optional.ofNullable(model.getSamples()).ifPresent(q -> {
+                q.stream().forEach(s -> ercf.process(s));
+                q.clear();
+            });
+            result = toResult(ercf.process(feature));
+        } else {
+            RandomCutForest rcf = model.getRcf();
+            ThresholdingModel threshold = model.getThreshold();
+            if (rcf == null || threshold == null) {
+                return new ThresholdingResult(0, 0, 0);
+            }
```

Comment:
For all of the other cases, instead of directly returning a result, you set the variable `result` and at the end of the function call:
```
modelState.setLastUsedTime(clock.instant());
return result;
```
Why is the return logic different here? Should we still make sure `modelState.setLastUsedTime(clock.instant())` is invoked before returning?

Reply:
This is the current behavior that doesn't need to be changed since the behavior is ok and this branch for rcf and thresholding will be shortly removed to leave ercf only.